quickwit/quickwit-serve/src/lib.rs

```diff
@@ -789,6 +803,19 @@ pub async fn serve_quickwit(
     Ok(actor_exit_statuses)
 }
 
+/// Stack of layers to use on the server side of the ingester service.
+fn ingester_service_layer_stack() -> IngesterServiceTowerLayerStack {
+    IngesterServiceClient::tower()
+        .stack_persist_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_open_replication_stream_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_init_shards_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_retain_shards_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_truncate_shards_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_close_shards_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_decommission_layer(quickwit_common::tower::OneTaskPerCallLayer)
+        .stack_layer(INGEST_GRPC_SERVER_METRICS_LAYER.clone())
```

Comment:
As I explained in another PR, we want `.stack_layer(INGEST_GRPC_SERVER_METRICS_LAYER.clone())` first.